### ISSUE TITLE
Unable to set rancher.network.interfaces.eth1.dhcp 

### ISSUE THREAD

**Issue by deniseschannon** [rancher/os#868]
Created based on forums post:
https://forums.rancher.com/t/setting-static-ip-address-in-rancheros-on-virtualbox-ros-set-command-does-nothing/1737

```
$ sudo ros config set rancher.network.interfaces.eth1.dhcp false
```
and then check for the result:
```
$ sudo ros config get rancher.network.interfaces.eth1.dhcp      
true
```
the overall eth1 config
```
$ sudo ros config export -f|grep eth1 -A5
      eth1:
        dhcp: true
        address: 192.168.99.101/24
        gateway: 0.0.0.0
        mtu: 1500
      lo:
```

**Comment by segfly** [comment]
@deniseschannon Are there any workarounds for this? I'm forced to use static IP address in my environment and I can't get rancheros network configuration to take effect. I have to manually set `ip addr` in order to get networking up and running - which is not persistent across reboots.

Thanks

**Comment by segfly** [comment]
Okay, I found my own workaround: If I just edit `/var/lib/rancher/config/cloud-config.yml` directly then the setting is effective.

**Comment by joshwget** [comment]
@segfly Which version of RancherOS are you using? I've been trying to reproduce this with v0.4.4 but I haven't had any luck so far.

**Comment by segfly** [comment]
@joshwget I was using v0.4.4 iso. I eventually moved to building a custom iso and putting my boot settings in the `os-config.tpl.yml` so I didn't have to type all this and more on a vmware console in my case.

I _think_ where I might have been going wrong with the `ros config set` was that I was trying to set `eth1`, but the actual interface was `eth0`. I realized this when I was editing the .yml file and so I never retried the CLI with the correct interface.

Thanks for looking into this.

**Comment by joshwget** [comment]
`ros config set` by itself doesn't do much other than update a cloud config file on disk. Even though the setting is related to networking, the issue here seems to just be in the former process. Since that would likely be a bigger issue, I'm definitely interested in being able to reproduce this :smile:

**Comment by segfly** [comment]
@joshwget 
I can't reproduce it now. I think it was a case of fat fingers. I either did `get` when I should have `set`, or I  was setting `eth1` and getting `eth0` or vice-versa. When I searched and found this github issue, it offered some confirmation bias that I was experiencing some kind of bug.

```
[rancher@rancher ~]$ sudo ros config set rancher.network.interfaces.eth1.dhcp false
[rancher@rancher ~]$ sudo ros config get rancher.network.interfaces.eth1.dhcp
false
```

Regardless, it would be great if `ros config` had more intelligence about the environment. Say, to warn you that you are trying to configure an interface that does not exist on the host.  :smiley: 
 
Thanks!

**Comment by joshwget** [comment]
I was finally able to reproduce this (in v0.4.4, too). This only seems to happen once RancherOS is installed to disk, which is why it was working fine for me earlier.

**Comment by segfly** [comment]
@joshwget
Glad to hear I did not imagine this. Thanks! :+1: 
I did indeed have RancherOS installed to disk at one point - which is likely when I had seen this.

**Comment by deniseschannon** [comment]
Verified fixed in v0.4.5-rc1
